spec/models/teacher_spec.rb

```diff
@@ -8,4 +8,16 @@
     it { is_expected.to validate_presence_of(:email) }
     it { is_expected.to validate_presence_of(:password) }
   end
+
+  describe 'email validation fail' do
```

Comment:
Congratulations for covering this edge case! Also, great use of `attributes_for` helper, I'm so proud ❤️ 

The only thing that I'd correct here are descriptions:

```suggestion
  describe 'email validation' do
```

You're describing email validation as whole, no need to dive whether the examples will be successful or not. In my opinion testing AR validations is one of the valid exceptions from the [better specs guidelines about `describe`](https://www.betterspecs.org/#describe)

Reply:
Check :) Thank You for reminding me about better_specs site.